src/ucs/sys/sock.c

```diff
@@ -485,3 +486,80 @@ int ucs_sockaddr_cmp(const struct sockaddr *sa1,
     }
     return result;
 }
+
+int ucs_sockaddr_is_inaddr_any(struct sockaddr *addr)
+{
+    switch (addr->sa_family) {
+    case AF_INET:
+        return UCS_SOCKET_INET_ADDR(addr).s_addr == INADDR_ANY;
+    case AF_INET6:
+        return !memcmp(&(UCS_SOCKET_INET6_ADDR(addr)), &in6addr_any,
+                       sizeof(UCS_SOCKET_INET6_ADDR(addr)));
+    default:
+        ucs_debug("Invalid address family: %d", addr->sa_family);
+    }
+
+    return 0;
+}
+
+ucs_status_t ucs_sockaddr_get_dev_names(unsigned *num_resources_p, 
```

Comment:
this function seems to belong to UCT/tcp/base rather than UCS
so it could return uct resources array

Reply:
will move to `uct/tcp/base`

Reply:
@yosefe At the moment, there isn't a directory named `uct/tcp/base`. On top of that, all files under `uct/tcp` are already named `tcp_*`. Shall I just move this function to `uct/tcp/tcp_iface.[ch]` and use it sockcm later?

Reply:
okay, pls move to tcp and include it from sockcm